### DISCUSSION TITLE
Is it possible to have scoped packages? (i.e. start with @)

### DISCUSSION THREAD

**Discussion by mmJordanO** [vercel/turborepo#315] [Help] [answered]
In the packages folder, I was able to get something without the `@` preceding it working, however when I try to do it with it it comes up as not found.

I have:
`/packages/@my-package/config`

Within that folder, the package.json defines the name the same way.  "@my-package/config"

Lastly, in `/packages/config/eslint-preset.js` I tried adding the new folder a few ways, but none worked.

Examples:
```
"packages/@my-package/"
"packages/@my-package/**"
"packages/@my-package/config/"
```

Am I missing something or can we not do this just yet?

**Comment by JeroenReumkens** [comment, marked as answer]
@mmJordanO  Yes, scoped packages seem possible! 
What you need to do is the following:

1. Make sure you update the name of your packages in `package.json` to contain the scopes (`@my-package/...`)
2. Don't change anything to the folder names — so `@my-package/foo` should be in the `packages/foo` directory
3. Run `yarn` and `yarn dev` again. Your packages should work now. 

If you also update the config packages, you also only change the names in the `package.json`. Make sure to replace all the references in other config files to these packages as well (linting, etc).

**Reply by mmJordanO** [reply]
Hm.  I can get it to work it the package folder name isn't preceded by an `@` symbol.  I guess it isn't make or break if there's no `@`...

Thanks for the response.

**Reply by yuchant** [reply]
> @mmJordanO Yes, scoped packages seem possible! What you need to do is the following:
> 
> 1. Make sure you update the name of your packages in `package.json` to contain the scopes (`@my-package/...`)
> 2. Don't change anything to the folder names — so `@my-package/foo` should be in the `packages/foo` directory
> 3. Run `yarn` and `yarn dev` again. Your packages should work now.
> 
> If you also update the config packages, you also only change the names in the `package.json`. Make sure to replace all the references in other config files to these packages as well (linting, etc).

I'm sorry could you say this one more time?

So package.json should contain a dependency with: "@my-package/foo"
The package should be "packages/my-package/foo" ? 

Thank you!

**Reply by JeroenReumkens** [reply]
@yuchant  No you don't need the `my-package` directory. Just `packages/foo`. 

```
packages
│
└───foo
               — package.json (name: @my-package/foo)
apps
│
└───app-1
               — package.json (name: @my-package/app-1)
               — index.js (import foo from @my-package/foo) 
```

**Reply by mmJordanO** [reply]
I see.  I can get that to work, but is there a way to do it like so?

```
my-package
│
└───a - package.json name is @my-package/a
│
└───b - package.json name is @my-package/b
```

Effectively, I want to divide my packages into a few subfolders with the same parent folder.  Basically, have the two package.json files set up and when I do a yarn install, everything gets grouped into that folder.

**Reply by JeroenReumkens** [reply]
Well, I don't know any specifics on Turborepo (I'm not a contributor or anything like that), but I can imagine that it could work if you update the `workspaces` property in the root `package.json` to include your directory as well. Something like;

```
"workspaces": [
  "apps/*",
  "packages/@my-package/*"
]
```

Might be that you already tried this of course 😁

**Reply by mmJordanO** [reply]
I didn't try that, specifically, but I would've thought that `packages/*` would just include everything automatically.

**Reply by JeroenReumkens** [reply]
No that will only include one level deep. You can probably also replace it with `packages/**/*`.

**Reply by mmJordanO** [reply]
Hm.  I did get something like this to work:

```
"workspaces": [
  "apps/*",
  "packages/my-package/*",
  "packages/*"
]
```

Kind of like special treatment for that one folder.  I can get it to work without scoping (i.e. yarn install builds it and can deploy a test to Vercel). In the `my-package` folder there's a package.json that builds the two subpackages in there.  The only downside I can see to that is needed to add onto the build command for any new package.

The perfect solution would be to have scoping and then anything added to my-package just gets built.

The workspaces addition has gotten me closer though.  Thanks for that.  :)

**Reply by bherbruck** [reply]
Do you think this would work with `npm` I switched my packages to have scoped `@` names in `package.json`. Now when I try to install a remote dependency in another package it is looking for my local packages in the npm registry! If I use names with out the scoped part, it works just fine.
